pkg/kubelet/dockertools/manager.go

```diff
@@ -1579,6 +1611,16 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 			glog.Errorf("Failed to create pod infra container: %v; Skipping pod %q", err, podFullName)
 			return err
 		}
+
+		// Find the pod IP after starting the infra container in order to expose
+		// it safely via the downward API without a race
```

Comment:
... the network plugin should update the pod IP we retrieved from runContainerInPod if and only if there is a non -default network plugin in use.